dir.targets

```diff
@@ -57,6 +57,14 @@
     
     <Error Condition="'$(ErrorIfBuildToolsRestoredFromIndividualProject)'=='true'"
            Text="The build tools package was just restored and so we cannot continue the build of an individual project because targets from the build tools package were not able to be imported. Please retry the build the individual project again." />
+
+    <!--
+      There are cases where the inputs could be newer than the outputs but the
+      download or restore may not need to update. In such cases we need to touch
+      these files otherwise we continually run this target over and over for
+      every project until these files are cleaned (if the are ever cleaned).
+    -->
+    <Touch Files="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll;$(NugetToolPath)" />
```

Comment:
So the problem is if you remove a dependency, or if you update the build system itself?